### PR TITLE
Documentation site dependency update

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

1. Update to latest Docusaurus `3.4.0`
2. Updated React `18.2.0`
3. Fixed issues with `prism-react-render` themes in `docusaurus.config.js`

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

Within `docs-site` directory:

1. `pnpm clear`
2. `pnpm install`
3. `pnpm build`
4. `pnpm start`

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: docs-site/package.json [hunk 1]
```diff
@@ -14,16 +14,17 @@
     "write-heading-ids": "docusaurus write-heading-ids"
   },
   "dependencies": {
-    "@docusaurus/core": "2.4.1",
-    "@docusaurus/preset-classic": "2.4.1",
+    "@docusaurus/core": "3.4.0",
+    "@docusaurus/preset-classic": "3.4.0",
     "@mdx-js/react": "^1.6.22",
```
Comment (ericenns) [inline docs-site/package.json:19]: **nitpick (blocking):** Can you update this to `^3.0.1`

File: docs-site/package.json [hunk 2]
```diff
@@ -14,16 +14,17 @@
...
     "clsx": "^1.2.1",
```
Comment (ericenns) [inline docs-site/package.json:20]: **nitpick (blocking):** Can you update this to `^2.1.1`

File: docs-site/package.json [hunk 3]
```diff
@@ -38,7 +39,7 @@
     ]
   },
   "engines": {
-    "node": ">=16.14"
+    "node": ">=20.14.0"
   },
   "pnpm": {
```
Comment (ericenns) [inline docs-site/package.json:44]: **nitpick (blocking):** Can you try removing this section, not sure but you may need to update to the latest version of pnpm in `.tool-versions`.

**Review by ericenns — CHANGES_REQUESTED** [review]
I cannot get it to build locally. I gave some suggestions on updates which I think should resolve my dependency install issue.

File: docs-site/package.json (hunk 1 above)
Comment (joshsadam) [reply docs-site/package.json]: Updated in 276320f4653f589781ce3dfc58ec9af8a1aec1e0

File: docs-site/package.json (hunk 2 above)
Comment (joshsadam) [reply docs-site/package.json]: Updated in 276320f4653f589781ce3dfc58ec9af8a1aec1e0

File: docs-site/package.json (hunk 3 above)
Comment (joshsadam) [reply docs-site/package.json]: Updated in 276320f4653f589781ce3dfc58ec9af8a1aec1e0

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**